ateam_common/src/multicast_receiver.cpp

```diff
@@ -32,16 +37,25 @@ namespace ateam_common
 MulticastReceiver::MulticastReceiver(
   std::string multicast_address_string,
   uint16_t multicast_port,
-  ReceiveCallback receive_callback)
+  ReceiveCallback receive_callback,
+  std::string interface_address)
 : receive_callback_(receive_callback),
   multicast_socket_(io_service_)
 {
-  const auto multicast_address = boost::asio::ip::make_address(multicast_address_string);
+  const auto multicast_address = boost::asio::ip::make_address(multicast_address_string).to_v4();
   const boost::asio::ip::udp::endpoint multicast_endpoint(multicast_address, multicast_port);
   multicast_socket_.open(multicast_endpoint.protocol());
   multicast_socket_.set_option(boost::asio::ip::udp::socket::reuse_address(true));
   multicast_socket_.bind(multicast_endpoint);
-  multicast_socket_.set_option(boost::asio::ip::multicast::join_group(multicast_address));
+  if (interface_address.empty()) {
+    multicast_socket_.set_option(boost::asio::ip::multicast::join_group(multicast_address));
+  } else {
+    multicast_socket_.set_option(
+      boost::asio::ip::multicast::join_group(
+        multicast_address,
+        boost::asio::ip::make_address_v4(interface_address)));
+  }
+  // JoinMulticastGroupOnAllV4Interfaces(multicast_address);
```

Comment:
We should remove this (and the function definitions below) since I never got it working at comp. Can come back around and try it again if we want to in the future.

```suggestion
```

Reply:
Can you mark the other functions here that need to be removed/edit this file? I'm not sure if you mean everything below line 58 here or something else based on your comment here.

Reply:
Removed this and the corresponding function definition.

Reply:
Done!